File: app/src/main/java/com/shveed/cookmegood/data/NetworkService.java
package com.shveed.cookmegood.data;

import retrofit2.Retrofit;
import retrofit2.converter.jackson.JacksonConverterFactory;

public class NetworkService {
    private static NetworkService mInstance;

    private static final String BASE_URL = "http://localhost:8080/";

    private Retrofit mRetrofit;

    private NetworkService() {
        mRetrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(JacksonConverterFactory.create())
                .build();
    }

    public static NetworkService getInstance() {
        if (mInstance == null) {
            mInstance = new NetworkService();
        }
        return mInstance;
    }

    public UserApi getUserApi() {
        return mRetrofit.create(UserApi.class);
    }

    public RecipeApi getRecipeApi(){
        return mRetrofit.create(RecipeApi.class);
    }


}

File: app/src/main/java/com/cookMeGood/makeItTasteIt/data/NetworkService.kt
package com.cookMeGood.makeItTasteIt.data

import retrofit2.Retrofit
import retrofit2.converter.jackson.JacksonConverterFactory

class NetworkService() {
    private val mRetrofit: Retrofit
        private var mInstance: NetworkService? = null
        private  val BASE_URL = "http://localhost:8080"

    init {
        mRetrofit = Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(JacksonConverterFactory.create())
                .build()
    }

    val instance: NetworkService?
        get() {
            if (mInstance == null) {
                mInstance = NetworkService()
            }
            return mInstance
        }

    val userApi: UserApi
        get() = mRetrofit.create(UserApi::class.java)

    val recipeApi: RecipeApi
        get() = mRetrofit.create(RecipeApi::class.java)

    val categoryApi: CategoryApi
        get() = mRetrofit.create(CategoryApi::class.java)
}